server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -417,6 +441,24 @@ public static void updateIndexSettings(
         }
     }
 
+    /**
+     * Verifies that read-only compatible indices always have a write block.
+     *
+     * @param readOnlyIndices the read-only compatible indices
+     * @param blocks the updated cluster state blocks
+     */
+    private static void verifyReadOnlyIndices(@Nullable Set<Index> readOnlyIndices, ClusterBlocks blocks) {
+        if (readOnlyIndices != null) {
+            for (Index readOnlyIndex : readOnlyIndices) {
+                if (blocks.indexBlocked(ClusterBlockLevel.WRITE, readOnlyIndex.getName()) == false) {
```

Comment:
Should this use the new `hasIndexBlockLevel` method to avoid interaction from global blocks?

Reply:
Right, I pushed de6a0bdb02283850083ed6d633577643dd5f33fb.